packages/editor/src/editor/canvas/ComponentBlock.tsx

```diff
@@ -78,7 +88,16 @@ const Draggable = ({ config, data }: DraggableProps) => {
           {config.render({ ...elementConfig, id: data.id })}
         </div>
       </PopoverAnchor>
-      <Quickbar deleteAction={deleteElement} duplicateAction={duplicateElement} createAction={createElement} />
+      {data.type === 'DataTableColumn' ? (
+        <Quickbar deleteAction={deleteElement} />
+      ) : (
+        <Quickbar
+          deleteAction={deleteElement}
+          duplicateAction={duplicateElement}
+          createAction={createElement}
+          createColumnAction={data.type === 'DataTable' ? createColumn : undefined}
+        />
+      )}
```

Comment:
seems also wrong to me that the componentblock needs to check what actions a type has... This should be defined in the config of the component itself...

Reply:
-> new subtask